torchgeo/datasets/substation_seg.py

```diff
@@ -0,0 +1,170 @@
+"""This module handles the Substation segmentation dataset."""
+
+import os
+from typing import Any
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import download_url, extract_archive
+
+
+class SubstationDataset(NonGeoDataset):
+    """SubstationDataset is responsible for handling the loading and transformation of substation segmentation datasets.
```

Comment:
These lines are likely over the 88 character line length limit.

Can you include a URL to link to the homepage and a more detailed description of the dataset? See other dataset files as examples of what things we like to document.